### PR TITLE
Fixed KeyboardInterrupt hanging the asyncio test runner

### PR DESCRIPTION
**Pull request by agronholm**
## Changes

This fixes ctrl+c causing the anyio pytest plugin to hang if the test is currently awaiting something. This was caused by the test runner dropping exceptions on the floor.

## Checklist

If this is a user-facing code change, like a bugfix or a new feature, please ensure that
you've fulfilled the following conditions (where applicable):

- [X] You've added tests (in `tests/`) added which would fail without your patch
- [ ] You've updated the documentation (in `docs/`, in case of behavior changes or new
features)
- [X] You've added a new changelog entry (in `docs/versionhistory.rst`).

If this is a trivial change, like a typo fix or a code reformatting, then you can ignore
these instructions.

### Updating the changelog

If there are no entries after the last release, use `**UNRELEASED**` as the version.
If, say, your patch fixes issue #123, the entry should look like this:

`* Fix big bad boo-boo in task groups (#123
<https://github.com/agronholm/anyio/issues/123>_; PR by @yourgithubaccount)`

If there's no issue linked, just link to your pull request instead by updating the
changelog after you've created the PR.

### CODE REVIEW

**Review by Zac-HD — COMMENTED** [review]
Implementation looks reasonable to me, though obviously the test failures aren't ideal.

I'll unsubscribe for now but do @-me or re-request review when that'd be helpful!

File: src/anyio/_backends/_asyncio.py [hunk 1]
```diff
@@ -2082,13 +2082,23 @@ async def _run_tests_and_fixtures(
             tuple[Awaitable[T_Retval], asyncio.Future[T_Retval]]
         ],
     ) -> None:
+        from _pytest.outcomes import OutcomeException
+
         with receive_stream, self._send_stream:
             async for coro, future in receive_stream:
                 try:
                     retval = await coro
+                except CancelledError:
```
Comment (graingert) [inline src/anyio/_backends/_asyncio.py:2091]: Do you want to extract and pass on the cancel message?

Comment (agronholm) [reply src/anyio/_backends/_asyncio.py:2091]: Hm, might be a good idea.

File: src/anyio/_backends/_asyncio.py [hunk 2]
```diff
@@ -2082,13 +2082,23 @@ async def _run_tests_and_fixtures(
             tuple[Awaitable[T_Retval], asyncio.Future[T_Retval]]
         ],
     ) -> None:
+        from _pytest.outcomes import OutcomeException
```
[the remark this answers is not in the document]
Comment (agronholm) [reply src/anyio/_backends/_asyncio.py:2085]: I'm not understanding. This is an import in the AnyIO pytest runner. Why would someone try to use that without pytest in the first place?

[the remark this answers is not in the document]
Comment (Zac-HD) [reply src/anyio/_backends/_asyncio.py:2085]: Ah, right, I just wasn't looking at a wide enough context 🤦

**Review by Zac-HD — APPROVED** [review]

**Comment by agronholm** [comment]
Thanks for the review!

**Pull request merged**